test/lib/test_services.hh

```diff
@@ -76,4 +79,18 @@ struct table_for_tests {
     future<> stop_and_keep_alive() {
         return stop().finally([cf = *this] {});
     }
+
+    sstables::shared_sstable make_sstable(std::optional<sstables::sstable_version_types> version = std::nullopt) {
+        auto& table = *_data->cf;
+        auto& sstables_manager = table.get_sstables_manager();
+        return sstables_manager.make_sstable(_data->s, _data->cfg.datadir, table.calculate_generation_for_new_table(), version.value_or(_data->sstables_version));
+    }
+
+    std::function<sstables::shared_sstable()> get_creator() {
+        return [this] { return make_sstable(); };
+    }
+
+    operator std::function<sstables::shared_sstable()>() {
```

Comment:
Please, add a comment that this thing is to make table_for_tests be useable as the factory argument to make_sstable_containing

Reply:
I'm actually getting rid of the cast to function in the next version, since I don't like it myself, in hindsight.
Instead, I'm introducing a `make_sst_factory` method to sstable_test_env and to table_for_tests which will be clearer as to what's going on.